pkg/controller/registry/resolver/resolver.go

```diff
@@ -55,6 +55,46 @@ func (w *debugWriter) Write(b []byte) (int, error) {
 	return n, nil
 }
 
+// DumpCache writes out the contents of the cache to logs
+// This is useful for debugging resolution errors
+func (r *Resolver) DumpCache(namespaces []string) {
+	if r.log == nil {
+		return
+	}
+
+	if r.cache == nil {
+		r.log.Warningln("resolver cache is nil")
+		return
+	}
+
+	cacheEntries := r.cache.Namespaced(namespaces...).Find(cache.True())
```

Comment:
Would it be more useful to print the same cache view that was used when resolution failed? I am concerned about wild goose chases stemming from logs containing the failure followed immediately by a cache dump that may not represent the inputs to the failure.